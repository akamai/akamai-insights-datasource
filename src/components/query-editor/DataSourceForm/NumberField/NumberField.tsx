import { InlineField, Input } from '@grafana/ui';
import React, { FormEvent, JSX } from 'react';

import { shortLabelWidth } from '../FormTypes';

export interface NumberFieldProps {
  fieldLabel: string;
  fieldTooltip?: any;
  inputWidth?: number;
  model: number | undefined;
  onChange: (value: number | undefined) => void;
  min?: number;
  max?: number;
}

export function NumberField({ fieldLabel, fieldTooltip, inputWidth, model, min, max, onChange }: NumberFieldProps): JSX.Element {
  const onInputChange = (event?: FormEvent<HTMLInputElement>) => {
    const value = Number((event?.target as any).value);
    onChange(!isNaN(value) ? value : undefined);
  };

  return (
    <InlineField
      label={fieldLabel}
      labelWidth={shortLabelWidth}
      tooltip={fieldTooltip}>
      <Input
        type="number"
        min={min}
        max={max}
        value={model}
        width={inputWidth}
        onChange={event => onInputChange(event)}
      />
    </InlineField>
  );
}
