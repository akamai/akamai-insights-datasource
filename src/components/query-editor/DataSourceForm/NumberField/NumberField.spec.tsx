import React from 'react';
import renderer from 'react-test-renderer';

import { NumberField } from './NumberField';

describe('NumberField component', () => {
  it('should render field', () => {
    const model = 100;

    const onChange = jest.fn();

    const tree = renderer.create(
      <NumberField fieldLabel="Limit" model={model} min={0} onChange={onChange}/>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
