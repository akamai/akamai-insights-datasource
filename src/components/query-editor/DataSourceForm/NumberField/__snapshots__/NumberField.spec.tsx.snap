// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NumberField component should render field 1`] = `
<div
  className="css-1f5gueu"
>
  <label
    className="css-1e5y6i9"
  >
    Limit
  </label>
  <div
    className="css-1rfunm5"
  >
    <div
      className="css-xcstkt-input-wrapper"
      data-testid="input-wrapper"
    >
      <div
        className="css-1w5c5dq-input-inputWrapper"
      >
        <input
          className="css-1n6y0bv-input-input"
          min={0}
          onChange={[Function]}
          style={
            {
              "paddingLeft": undefined,
              "paddingRight": undefined,
            }
          }
          type="number"
          value={100}
        />
      </div>
    </div>
  </div>
</div>
`;
